File: Easy/1071. Greatest Common Divisor of Strings/Solution.cs
using System;

public class Solution {
    public string GcdOfStrings(string str1, string str2) {
        if (str1 + str2 != str2 + str1) {
            return "";
        }
        
        int gCD = FindGCDByEuclidean(str1.Length, str2.Length);
        
        return str1.Substring(0, gCD);
    }

    private static int FindGCDByEuclidean(int a, int b) {
        if (b == 0) {
            return a;
        }
        return FindGCDByEuclidean(b, a % b);
    }
}

File: Easy/1071. Greatest Common Divisor of Strings/Solution.java
class Solution {
    public String gcdOfStrings(String str1, String str2) {
        if (!((str1 + str2).equals(str2 + str1))) {return "";}
        int gCD = findGCDByEuclidean(str1.length(),str2.length());
        
        return str1.substring(0,gCD);
    }

    public static int findGCDByEuclidean(int a, int b) {
        if (b == 0) {
            return a;
        }
        return findGCDByEuclidean(b, a % b);
    }
}
